hcn/hcnsupport.go

```diff
@@ -43,7 +50,59 @@ type ApiSupport struct {
+	// debug incidents where there's confusion on if a feature is supported on the host machine. The sync.Once helps to avoid redundant
+	// spam of these anytime a check needs to be made for if an HCN feature is supported. This is a common occurrence in kube-proxy
+	// for example.
+	featuresOnce.Do(func() {
+		globals, err := GetGlobals()
+		if err != nil {
+			// It's expected if this fails once, it should always fail. It should fail on pre 1803 builds for example.
+			versionErr = errors.Wrap(err, "failed to query HCN version number: this is expected on pre 1803 builds.")
+		} else {
+			supportedFeatures.Acl = AclFeatures{
+				AclAddressLists:       isFeatureSupported(globals.Version, HNSVersion1803),
+				AclNoHostRulePriority: isFeatureSupported(globals.Version, HNSVersion1803),
+				AclPortRanges:         isFeatureSupported(globals.Version, HNSVersion1803),
+				AclRuleId:             isFeatureSupported(globals.Version, HNSVersion1803),
+			}
+
+			supportedFeatures.Api = ApiSupport{
+				V2: isFeatureSupported(globals.Version, V2ApiSupport),
+				V1: true, // HNSCall is still available.
+			}
+
+			supportedFeatures.RemoteSubnet = isFeatureSupported(globals.Version, RemoteSubnetVersion)
+			supportedFeatures.HostRoute = isFeatureSupported(globals.Version, HostRouteVersion)
+			supportedFeatures.DSR = isFeatureSupported(globals.Version, DSRVersion)
+			supportedFeatures.Slash32EndpointPrefixes = isFeatureSupported(globals.Version, Slash32EndpointPrefixesVersion)
+			supportedFeatures.AclSupportForProtocol252 = isFeatureSupported(globals.Version, AclSupportForProtocol252Version)
+			supportedFeatures.SessionAffinity = isFeatureSupported(globals.Version, SessionAffinityVersion)
+			supportedFeatures.IPv6DualStack = isFeatureSupported(globals.Version, IPv6DualStackVersion)
+			supportedFeatures.SetPolicy = isFeatureSupported(globals.Version, SetPolicyVersion)
+			supportedFeatures.VxlanPort = isFeatureSupported(globals.Version, VxlanPortVersion)
+			supportedFeatures.L4Proxy = isFeatureSupported(globals.Version, L4ProxyPolicyVersion)
+			supportedFeatures.L4WfpProxy = isFeatureSupported(globals.Version, L4WfpProxyPolicyVersion)
+			supportedFeatures.TierAcl = isFeatureSupported(globals.Version, TierAclPolicyVersion)
+			supportedFeatures.NetworkACL = isFeatureSupported(globals.Version, NetworkACLPolicyVersion)
+			supportedFeatures.NestedIpSet = isFeatureSupported(globals.Version, NestedIpSetVersion)
+
+			logrus.WithFields(logrus.Fields{
+				"version":           fmt.Sprintf("%+v", globals.Version),
+				"supportedFeatures": fmt.Sprintf("%+v", supportedFeatures),
+			}).Info("HCN feature check")
+		}
+	})
+
+	return supportedFeatures, versionErr
+}
+
+// GetSupportedFeatures returns the features supported by the Service. Prefer `GetCachedSupportedFeatures` as this method will query hns and validate
+// every feature is supported on every invocation.
 func GetSupportedFeatures() SupportedFeatures {
```

Comment:
We should mark this function as deprecated (add a comment like `// Deprecated: Use GetCachedSupportedFeatures instead.`).